src/sentry/notifications/platform/target.py

```diff
@@ -49,6 +65,17 @@ class GenericNotificationTarget(NotificationTarget):
     If all targets use the same payload, please add this to the NotificationTemplate instead.
     """
 
+    def to_dict(self) -> dict[str, Any]:
+        return {
+            **asdict(self),
+        }
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> Self:
+        if "type" in data:
+            data.pop("type")
+        return cls(**data)
```

Comment:
This is coupled a bit too closely to the serialized typeDict (which we could turn into a dataclass to handle some of this type conversion I think)

Reply:
yeah I think we can insert this in the wrapper class below to handle specific serializations for the task